Tool/SuperCAT.ps1

```diff
@@ -1,400 +1,345 @@
+    if ($PassThru) {
+        return $Config
     }
     else{
-        .$Drive\AV\w32\SCAN /DRIVER=$Drive\AV\DAT /ANALYZE /ADL /SECURE /NOBREAK /TIMEOUT=10 /THREADS=64 /REPORT=$AVLogs\$ComputerName-AV-Report.txt /HTML $AVLogs\$ComputerName-AVREPORT.html
+        return $True
     }
 }
-
-###########################################################################################################
-## Option 3 sends any available antivirus logs to the location of the script (i.e. CD).
-###########################################################################################################
-
-if((($Choices -Contains 3) -or ($Choices -Contains 6)) -and ($Choices -NotContains 7)){
-    # Gathers scan logs already on the machine.
-    Write-Output "Gathering scan logs..."
-    if($Win32Caption -like "*Windows 7*"){
-        Copy-Item -Path "C:\ProgramData\McAfee\DestkopProtection\OnDemandScanLog.txt" -Destination "$AVLogs"
-    }
-    else{
-        Copy-Item -Path "C:\ProgramData\McAfee\Endpoint Security\Logs\" -Destination "$AVLogs" -Recurse
+function Get-LogPrefix {
+    # .SYNOPSIS
+    # Pull system abbreviations and mesh config data to provide a log prefix.
+    param (
+        [Parameter(Mandatory=$True,ValueFromPipeline=$True,Position=0)]
+        [PSCustomObject]$Config,
+
+        [Parameter(Mandatory=$True,Position=1)]
+        [ValidateNotNullOrEmpty()]
+        [String]$RootDirectory,
+
+        [Parameter()]
+        [Switch]$SCAP
+    )
+    Try {
+        $PlatformList = Import-CSV -Path "$RootDirectory\PlatformList.csv"
+    } Catch {
+        throw "PlatformList.csv doesn't exist! Please reinstall SuperCAT."
     }
-}
 
-###########################################################################################################
-## Option 4 runs the DISA SCAP tool to check on the system's STIG compliance, send it back for analysis.
-###########################################################################################################
-
-if((($Choices -Contains 4) -or ($Choices -Contains 6)) -and ($Choices -NotContains 7)){
-    # Conducts a SCAP check on the machine and writes it to a file.
-    if(test-path "$Drive\DISA\cscc.exe"){
-        Write-Output "Running SCAP Scan..."
-        & $Drive\DISA\cscc.exe -u $SCAPLogs
+    $SystemAbbreviation = $(
+        $PlatformList | Where-Object {
+            $_.SystemName -eq $Config.KnownDrives.$($Config.LastHDD).SystemType
+        }).SystemAbbreviation
+    $SystemOwner  = $Config.KnownDrives.$($Config.LastHDD).SystemOwner
+    $Date         = $(Get-Date -Format "yyyyMMdd_HHmm" -Date $Config.LastAccessTimeUTC)
+    $Version      = $Config.KnownDrives.$($Config.LastHDD).SystemVersion
+    $Drive        = $Config.KnownDrives.$($Config.LastHDD).DriveName
+    if ($SCAP) {
+        return "$SystemOwner`_$SystemAbbreviation$Version`_$Drive"
+    } else {
+        return "$Date`_$SystemOwner`_$SystemAbbreviation$Version`_$Drive"

```

Comment:
Agreed that we need to nail down the file name scheme. SystemOwner in this case is who owns/uses this HDD. Separate from ScanningOrg, which is the organization in charge of monitoring and analysis. SystemName(/Type), SystemAbbreviation, and Version should be pull-able from the system, just not in a format available on GitHub. This leaves SystemOwner and DriveName to user entry. The reason we used DriveName, which is actually a padded three digit number, is for brevity and consistency in the file name scheme. System owner could theoretically be a drop down, but that would require constant monitoring and updating.  The reason for not using $HardDriveModel_$HardDriveSerial is enabling instant comprehension by both security personnel and the system owners.

As an aside, SCAP has its own folder structure we're not messing with to allow for usage of existing analysis programs.

Reply:
"The reason for not using $HardDriveModel_$HardDriveSerial is enabling instant comprehension by both security personnel and the system owners."

I would that would drive us to use Model_SerialNumber instead, since those drives will (should) have a sticker on the outside of them that would correspond to the filename.

Is "SystemOwner" a new prompt that was added to the setup script part of the script? I haven't run your latest in a few days, but don't remember being prompted for that when I ran it.

Reply:
SystemOwner is the genericized Unit, but the meaning of the question is the same.  The thought behind DriveName is that the system owners have these numbered already for tracking purposes (i.e. Laptop 1 w/ Drive 5 checked out).